api-catalog-services/src/main/java/org/zowe/apiml/apicatalog/standalone/StandaloneLoaderService.java

```diff
@@ -0,0 +1,130 @@
+            log.error("No service definition files found.");
+            return;
+        }
+
+        for (File file : appFiles) {
+            createContainerFromFile(file);
+        }
+    }
+
+    private void loadOpenAPICache() {
+        File[] openAPIFiles = getFiles(servicesDirectory + "/apiDocs");
+        if (openAPIFiles.length == 0) {
+            log.error("No apiDocs files found.");
+            return;
+        }
+
+        for (File openAPIFile : openAPIFiles) {
+            loadApiDocCache(openAPIFile);
+        }
+    }
+
+    private void createContainerFromFile(File file) {
+        log.info("Initialising services from '{}' file.", file.getName());
+
+        try {
+            Applications apps = objectMapper.readValue(file, Applications.class);
+            apps.getRegisteredApplications().forEach(app -> {
+                instanceInitializeService.createContainers(app);
+                // Uses metadata from the first instance like {@link InstanceRetrievalService#getInstanceInfo}
+                loadApiDocVersionCache(app.getInstances().get(0));
+            });
+        } catch (IOException e) {
+            log.error("Unable to parse service definition '{}' because {}", file.getName(), e.getMessage());
+        }
+    }
+
+    private void loadApiDocCache(File file) {
+        try {
+            String apiDoc = IOUtils.toString(Files.newInputStream(file.toPath()), StandardCharsets.UTF_8);
+            String[] name = FilenameUtils.removeExtension(file.getName()).split("_");
+
+            if (name.length < 2) {
+                log.warn("ApiDoc file has incorrect format '{}'. The correct format is 'serviceId_version'.", apiDoc);
+                return;
+            }
+
+            if (name.length > 2 && name[2].equals("default")) {
+                cachedApiDocService.updateApiDocForService(name[0], CachedApiDocService.DEFAULT_API_KEY, apiDoc);
+            }
+
+            cachedApiDocService.updateApiDocForService(name[0], name[1], apiDoc);
+        } catch (IOException e) {
+            log.error("Cannot read '{}' because {}", file.getName(), e.getMessage());
+        }
+    }
+
+    private void loadApiDocVersionCache(InstanceInfo instanceInfo) {
+        List<String> apiVersions = standaloneAPIDocRetrievalService.retrieveApiVersions(instanceInfo.getMetadata());
+        cachedApiDocService.updateApiVersionsForService(instanceInfo.getAppName(), apiVersions);
+
+        String defaultApiVersion = standaloneAPIDocRetrievalService.retrieveDefaultApiVersion(instanceInfo.getMetadata());
+        cachedApiDocService.updateDefaultApiVersionForService(instanceInfo.getAppName(), defaultApiVersion);
+    }
+
+    private File[] getFiles(String directory) {
+        File dir = new File(directory);
+        if (!dir.isDirectory()) {
+            log.error("Directory '{}' does not exists.", directory);
+            return new File[0];
+        }
+
+        return dir.listFiles((d, name) -> name.endsWith(".json"));
```

Comment:
There should be a check of type (file/dir) as well.
```suggestion
        return dir.listFiles((f, name) -> f.isFile() && name.endsWith(".json"));
```

Reply:
It does not work. `d` is the original directory, not a file.